mlflow/entities/span_status.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from opentelemetry import trace as trace_api
+
+
+@dataclass
+class SpanStatus:
+    """
+    Status of the span or the trace.
+
+    Args:
+        status_code: The status code of the span or the trace.
+        description: Description of the status. Optional.
+    """
+
+    status_code: StatusCode
+    description: str = ""
+
+    class StatusCode:
+        # Importing here to avoid circular imports
+        from mlflow.entities.trace_status import TraceStatus
+
+        UNSPECIFIED = TraceStatus.to_string(TraceStatus.UNSPECIFIED)
```

Comment:
Note: `TraceStatus` and `SpanStatus` shares the same set of status code, but differs in the existence of `description` field. Not doing `status_code: TraceStatus` because it may be a bit confusing to see sth like `status = SpanStatus(TraceStatus.OK, description="...")`.